MaxText/train.py

```diff
@@ -211,13 +212,30 @@ def train_loop(config, state=None):
   Returns:
 
   """
+
+  monitoring_enabled = config.enable_cloud_monitoring
+
+  if monitoring_enabled:
```

Comment:
let's move registrations and all this logic into a standalone function. register_all_train_metrics